test/e2e/apimachinery/resource_quota.go

```diff
@@ -1186,21 +1186,36 @@ var _ = SIGDescribe("ResourceQuota", func() {
 		})
 		framework.ExpectNoError(err, "failed to locate ResourceQuota %q in namespace %q", patchedResourceQuota.Name, ns)
 
-		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 5*time.Minute, true, func(ctx context.Context) (bool, error) {
+		// the resource_quota_controller ignores changes to the status so we have to wait for a full resync of the controller
+		// to reconcile the status again, this full resync is set every 5 minutes by default so we need to poll at least one
+		// minute more just in case we we start to poll just after the full resync has happened and he have to wait until
+		// next full resync.
+		// Ref: https://issues.k8s.io/121911
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 6*time.Minute, true, func(ctx context.Context) (bool, error) {
 			resourceQuotaResult, err := rqClient.Get(ctx, rqName, metav1.GetOptions{})
-			framework.ExpectNoError(err)
```

Comment:
we don't expect the get to error, right? when would that be expected / ok?

Reply:
this is inside the poll loop,  why should we fail hard here? usually we always retry in these cases

Reply:
The poll is to give the reconciler time to make the status value what we want. An unstable API server here is unexpected

Reply:
I  stand corrected, after a quick grep  I have to fully agree with you, I expected to see more retries on other tests, but it seems the norm is to fail here

Reply:
https://github.com/kubernetes/kubernetes/pull/121972

Reply:
if this GET API call must be one shot and not fail then this:
```go
resourceQuotaResult, err := rqClient.Get(ctx, rqName, metav1.GetOptions{})
framework.ExpectNoError(err)
```
must be outside of the poll loop.

Reply:
but you have to try to get everytime, or you don't get the new data, returning an error makes the same efect and avoid panicing inside https://github.com/kubernetes/kubernetes/pull/121972

Reply:
yes, and that's why this part of this PR, LGTM.

the change in https://github.com/kubernetes/kubernetes/pull/121972 is an improvement where the overall poll will know about the reported GET errors. that LGTM as well.